lib/attributor/attribute_resolver.rb

```diff
@@ -31,11 +30,16 @@ def query!(key_path, path_prefix=ROOT_PREFIX)
       _root, *path = key_path.split(SEPARATOR)
 
       # Follow the hierarchy path to the requested node and return it
-      # Example path => ["instance", "ssh_key", "name"]
-      # Example @data => {"instance" => { "ssh_key" => { "name" => "foobar" } }}
+      # Example path => ["instances", "at(0)", "ssh_key", "name"]
+      # Example @data => {"instances" => [{ "ssh_key" => { "name" => "foobar" } }]}
       result = path.inject(@data) do |hash, key|
         return nil if hash.nil?
-        hash.send key
+        # at(n) means we have a collection index
+        if (match = key.match(/^at\((\d+)\)$/))
```

Comment:
The regexp instance should be defined at the top of the class, frozen and used here.